src/main/java/jewellery/inventory/service/SaleService.java

```diff
@@ -252,17 +254,21 @@ private void throwExceptionIfResourceIsNotOwned(SaleRequestDto saleRequestDto) {
     }
   }
 
-  private List<PurchasedResourceInUser> getResourcesFromSaleRequestDto(
+  public List<PurchasedResourceInUser> getResourcesFromSaleRequestDto(
       SaleRequestDto saleRequestDto) {
-    List<PurchasedResourceInUser> resources = new ArrayList<>();
     if (saleRequestDto.getResources() != null) {
-      for (PurchasedResourceQuantityRequestDto resourceRequest : saleRequestDto.getResources()) {
-        PurchasedResourceInUser purchasedResourceInUser =
-            getPurchasedResourceInUser(resourceRequest);
-        resources.add(purchasedResourceInUser);
+      List<PurchasedResourceInUser> resources = new ArrayList<>();
+      if (saleRequestDto.getResources() != null) {
+        for (PurchasedResourceQuantityRequestDto resourceRequest : saleRequestDto.getResources()) {
+          PurchasedResourceInUser purchasedResourceInUser =
+              getPurchasedResourceInUser(resourceRequest);
+          resources.add(purchasedResourceInUser);
+        }
```

Comment:
This looks a bit weird. Why are we checking `if (saleRequestDto.getResources() != null)` twice?